tests/unit/asio_service_stream_test.cxx

```diff
@@ -88,54 +209,51 @@ namespace asio_service_stream_test {
             while (count > 0) {
                 ptr<req_msg> req(cs_new<req_msg>(
                     1, msg_type::append_entries_request, 1, my_id_,
-                    1, sent_log_index_, 1));
+                    1, stream_stat_->sent_log_index_, 1));
 
                 ptr<log_entry> log(cs_new<log_entry>(0, msg, log_val_type::app_log));
                 req->log_entries().push_back(log);
 
                 rpc_handler h = (rpc_handler)std::bind(
-                    &stream_server::handle_result,
+                    &stream_client::handle_result,
                     this,
                     req,
                     std::placeholders::_1,
                     std::placeholders::_2);
-                my_client_->send(req, h);
-                sent_log_index_++;
-                pp.update(sent_log_index_);
+                client_->send(req, h, timeout_ms);
+                stream_stat_->sent_log_index_++;
+                pp.update(stream_stat_->sent_log_index_);
                 count--;
             }
             pp.done();
-            num_messages_sent_= sent_log_index_;
+            stream_stat_->num_messages_sent_= stream_stat_->sent_log_index_;
         }
 
         void handle_result(ptr<req_msg>& req,
                            ptr<resp_msg>& resp,
                            ptr<rpc_exception>& err)
         {
-            if (resp->get_next_idx() == get_next_log_index()) {
-                next_log_index_++;
+            if (err) {
+                stream_stat_->error_req_count_++;
+                stream_stat_->next_log_index_++;
+                SimpleLogger* ll = stream_stat_->client_logger_->getLogger();
+                _log_info(ll, "handle result err: %s", err->what());
```

Comment:
It will be better to put `next_log_index_` and `error_req_count_` in the log for easier debugging.